### PR TITLE
[WIP][10.0] Update Openeducat and Fork

### PR DESCRIPTION
**Pull request by angelmoya**

### CODE REVIEW

**Comment by lasley** [comment]
Hmmmm I only see translation files here. Perhaps an issue with the format-patch

**Comment by angelmoya** [comment]
Now last commit is right, previous commits are from openeducat repository.

**Comment by pedrobaeza** [comment]
Did you talk with current OpenEducat maintainers (https://github.com/openeducat/openeducat_erp) to see if they want to merge their work here?

**Comment by angelmoya** [comment]
@pedrobaeza, few months ago I asked them about multi company support, and they told me that is supported on "enterprise version", this versión is not fully functional, there are a lot of models partially done.

Also I check on models some fields that are only for indian academic environments.

And also I want to integrate with other partner-contact modules.

I think that we can do best academic modules here.

**Comment by lasley** [comment]
I agree with @angelmoya - it's obvious they're pushing towards the Enterprise. There's absolutely no problem with that, but I see no problem with us continuing the fork either.

I think we will need to do something to avoid name conflicts between the modules though. Maybe add `oca_` prefix, although that seems kind of lame

File: education/demo/student_demo.xml [hunk 1]
```diff
@@ -348,7 +331,6 @@
             <field name="last_name">Bennett</field>
             <field name="gender">f</field>
```
Comment (lasley) [inline education/demo/student_demo.xml]: We have a `gender` in [`partner_contact_gender`](https://github.com/OCA/partner-contact/tree/10.0/partner_contact_gender), although we'd need to ask for a license change to include here.

File: education/__manifest__.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
         'partner_firstname',
         'partner_second_lastname',
         'partner_contact_birthdate',
+        'partner_contact_nationality'
```
Comment (lasley) [inline education/__manifest__.py]: These partner_contact modules are AGPL

Comment (lasley) [reply education/__manifest__.py]: We got around this in medical by creating an `_agpl` module that binds in the AGPL assets - https://github.com/LasLabs/vertical-medical/tree/release/10.0/medical_agpl

**Comment by angelmoya** [comment]
Hi @lasley , I update student and faculty models and views, to use partner-contact modules.

I will take a look to this _agpl module, I think that could be required here

**Comment by lasley** [comment]
> I will take a look to this _agpl module, I think that could be required here

Yeah unless we want to relicense to AGPL. I don't think that's a good idea though, AGPL has hurt me more than it's helped. Could just be me though.

**Comment by dreispt** [comment]
As a general rule, the code authors must have signed the OCA CLA.
Its important to confirm that (the authors of the code (a) are properly credited, and (b) agree with its inclusion in the OCA.

If not, we can still do a fork, but the README should have a disclaimer that this repor is not "clean", in the sense that parts of this code might not be under the OCA CLA.

**Comment by lasley** [comment]
Given the recent events in [Vertical Medical](https://github.com/OCA/vertical-medical/issues/183) that have arisen from the same source as this repo, we should tread very carefully here.

@angelmoya - how much work are you going to have to put into this? Is a rewrite that much more effort?

**Comment by dreispt** [comment]
I checked and it seems TechReceptives has signed the ECLA.
Still, I think it's polite to ask for permission.

**Comment by lasley** [comment]
I'm seeing similar issues with the IP in this project as I did in Vertical Medical, such as a copyright assertion of 2004 when the code looks to have been created in [2012 at the very earliest](http://bazaar.launchpad.net/~openeducat/openeducat/trunk/changes/10?start_revid=170). 

In another striking similarity to Vertical Medical - the project started with an abundance of files of which the development lineage is not totally clear. The first commit contains over 10k lines of code, with a commit message "changed repo structure".

Are we sure that this wasn't forked from somewhere else, such as what happened with GNUHealth? It's sad that I'm being this pessimistic here but IMO their copyright and licensing assertions are essentially worthless at this point in my eyes, and their code could be a liability.

That's my thoughts given my recent experiences in the very similar situation with these same authors. I don't know the history of these repos though, and it seems everyone other than me was aware of the GNUHealth thing.

With that said, I'll support this either way - I just felt the need to advise caution. I'm about 20 hours into a code audit that will likely kill 80 more, just to unravel a mess that I'm pretty sure my team coded us out of over a year ago because the original code was garbage.

**Comment by angelmoya** [comment]
Hi @lasley and @dreispt , I'm rewrite a lot of code, and clean a lot of unused code. I found a lot of fields with data focused on India Education System... and a lot of "merchandaising information"...

The core I think that is working propertly, and now I will update readme rst and tests for this PR.

After that I will do other PR with multicompany.

Maybe we could work on OCA Sprint Code on this.

**Comment by dreispt** [comment]
My personal opinion is that you either confirm with Techreceptives they agree contributing their code here, or start from scratch.

**Comment by angelmoya** [comment]
Hi @siddharth-techreceptives , I add you to this PR because you are the main contributor  of openeducat project.

Our goal is to migrate openeducat to OCA, and work on functionality like multicompany that is not on openeducat.

We would like to know your opinion about this work, and know if Techreceptives are agree to contributing their code here.

**Comment by JayVora-SerpentCS** [comment]
I believe its good to make a PR per module. Makes the review easier.

**Comment by angelmoya** [comment]
OK, I will do it

**Comment by dreispt** [comment]
@angelmoya I have licensing/copyright concerns about having code from Openeducat brought into the OCA without the Techreceptives explicit approval.
In general the GPL licenses allow you to do it, but for the OCA that is not enough: we need to be sure the contributed is covered bu the CLA. (cc @lasley )

**Comment by lasley** [comment]
I've already stated my concerns on this [here](https://github.com/OCA/vertical-education/pull/4#issuecomment-323447700). I have been burned by this entity before, even after I went through very extreme lengths to get explicit approval. Turns out the approval wasn't their's to give.

**Comment by lasley** [comment]
That said, we may just need to be clear that the repo isn't covered by standard OCA governance, and just adhere to GPL. I leave this up to @angelmoya seeing as he's doing the work, but we will need to put very very very clear documentation of this lack of governance.

**Comment by lasley** [comment]
It's worth noting in vertical-medical that I threw away at least 200 hours of work from my team just to get rid of the mess that was caused. Just rewriting code does not get you out from underneath the umbrella of bad licensing

(Sorry for the spam but I kept having more thoughts)

**Comment by angelmoya** [comment]
Hi @lasley  and @dreispt  we can talk about it next week on sprint code, I think that I don't understand all.

**Comment by angelmoya** [comment]
I cancel this PR because I will create new repository, without openeducat code.

**Comment by dreispt** [comment]
Did you have a chance to discuss this is person  with anyone?

**Comment by angelmoya** [comment]
We can try to talk, I send email with no answer, and base code is not to much

**Comment by lasley** [comment]
@angelmoya - Honestly I think you're making a good call here. Please let me know how I can help - I'm quite interested in this topic, and will definitely toss some dev effort in the pot now that we're going to have known good code.

**Comment by angelmoya** [comment]
@lasley  thanks a lot for your support, but finally I decide to start new module on version 11.
OpenEducat published code is only a piece of this project, some views are lost, there are web controller without views, a lot of fields useless.... So, as techreceptives is not receptive to mantain their project in OCA, I think is better to start a new project, I will start a new PR WIP for version 11

**Comment by bealdav** [comment]
@angelmoya you mean openeducat is not so open ? Interesting information - opensource washing then. 

I think this vertical could have a really good future all over the world. Thanks for the initiative to push it on v11.

**Comment by lasley** [comment]
I agree with @bealdav that there is potential here.

@angelmoya - do you have any sort of project plan or data outline or anything? I'd be willing to take some of this on as a personal thing.

**Comment by angelmoya** [comment]
Hi @lasley I have a project and requirement document, but in spanish, I will prepare a prototype in next two month, and production in 6 months.

I will do basics process, for academy in multicompany environment. And linked with crm and invoicing.

**Comment by lasley** [comment]
Nice, sounds like you're good then? Let me know when/if I can assist then 😄

**Pull request closed without merging**